bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,412 @@
+ 
+-        sed -i "/^\[vtpm\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR\/vtpm0#" $USER_HOME/.intel/.civ/civ-sriov.ini
++        sed -i "/^\[vtpm\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR\/$VM_NAME\/vtpm0#" $USER_HOME/.intel/.civ/civ-sriov.ini
+ 
+-        sed -i "/^\[rpmb\]$/,/^\[/ s#^bin_path.*#bin_path=$CIV_WORK_DIR\/scripts\/rpmb_dev#" $USER_HOME/.intel/.civ/civ-sriov.ini
+-        sed -i "/^\[rpmb\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR#" $USER_HOME/.intel/.civ/civ-sriov.ini
++        sed -i "/^\[rpmb\]$/,/^\[/ s#^bin_path.*#bin_path=$CIV_WORK_DIR\/$VM_NAME\/scripts\/rpmb_dev#" $USER_HOME/.intel/.civ/civ-sriov.ini
++        sed -i "/^\[rpmb\]$/,/^\[/ s#^data_dir.*#data_dir=$CIV_WORK_DIR\/$VM_NAME#" $USER_HOME/.intel/.civ/civ-sriov.ini
+ 
+-        sed -i "/^\[aaf\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR/scripts/aaf#" $USER_HOME/.intel/.civ/civ-sriov.ini
++        sed -i "/^\[aaf\]$/,/^\[/ s#^path.*#path=$CIV_WORK_DIR/$VM_NAME/scripts/aaf#" $USER_HOME/.intel/.civ/civ-sriov.ini
+         sed -i "/^\[aaf\]$/,/^\[/ s#^support_suspend.*#support_suspend=disable#" $USER_HOME/.intel/.civ/civ-sriov.ini
+ 
+         sed -i "/^\[passthrough\]$/,/^\[/ s/^passthrough_pci/#passthrough_pci/" $USER_HOME/.intel/.civ/civ-sriov.ini
++
++	mv $USER_HOME/.intel/.civ/civ-sriov.ini $USER_HOME/.intel/.civ/$VM_NAME.ini
+ }
+ 
+ function copy_civ_ini() {
+@@ -39,6 +48,30 @@ function copy_civ_ini() {
+         chmod 0666 $USER_HOME/.intel/.civ/civ-sriov.ini
+ }
+ 
++function read_args() {
++        while [[ $# -gt 0 ]]; do
++                case $1 in
++                        -v)
++                                VSOCK_ID=$2
++                                shift
++                                ;;
++
++                        -p)
++                                ADB_PORT=$2
++                                shift
++                                ;;
++
++                        -n)
++                                VM_NAME=$2
++                                shift
++                                ;;
++                esac
++                shift
++        done
++}
++
++read_args "$@"
++
+ copy_civ_ini
+ setup_civ_ini
+ 
+diff --git a/scripts/setup_host.sh b/scripts/setup_host.sh
+index 0c1e0a1..a4f72b9 100755
+--- a/scripts/setup_host.sh
++++ b/scripts/setup_host.sh
+@@ -19,6 +19,10 @@ CIV_WORK_DIR=$(pwd)
+ CIV_GOP_DIR=$CIV_WORK_DIR/GOP_PKG
+ CIV_VERTICAl_DIR=$CIV_WORK_DIR/vertical_patches/host
+ VM_MANAGER_VERSION=v1.2.3
++VSOCK_ID=3
++VM_NAME=Android-CIV
++ADB_PORT=5555
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
+ 
+ #---------      Functions    -------------------
+ function error() {
+@@ -575,6 +579,21 @@ function parse_arg() {
+                 setup_remote_infer || return -1
+                 ;;
+ 
++	    -v)
++                VSOCK_ID=$2
++		shift
++		;;
++
++	    -b)
++                ADB_PORT=$2
++		shift
++                ;;
++
++            -n)
++                VM_NAME=$2
++		shift
++                ;;
```

Comment:
Is it necessary to add these new arguments for ADB port and vsock in setup_host.sh?
Can we just use a default setting for adb port and vsock?
What is the usage scenario here?

Reply:
Removed vsock and adb ports. But vm-name is needed as it is needed if setup_host is being run from the new script for the first time.